api/utils/retryutils/retry.go

```diff
@@ -185,7 +186,8 @@ func (r *Linear) For(ctx context.Context, retryFn func() error) error {
 		if err == nil {
 			return nil
 		}
-		if _, ok := trace.Unwrap(err).(*permanentRetryError); ok {
+		var permanentRetryError *permanentRetryError
+		if errors.As(trace.Unwrap(err), &permanentRetryError) {
```

Comment:
Does this need the Unwrap here if we are using errors.As ?

Reply:
Similar thread at https://github.com/gravitational/teleport/pull/37669#discussion_r1475118680.

We don't need the unwraps, but I think Tim was avoiding changing too much.

Reply:
Ah - fair enough - I missed that comment since it was resolved 🙈